dbux-graph-host/src/graph/controllers/FocusController.js

```diff
@@ -1,184 +1,56 @@
+  handleTraceSelected = async () => {
+    if (!this.context.graphContainer.isEnabled()) {
+      return;
+    }
+
     const trace = traceSelection.selected;
     try {
       await this.waitForInit();
-      if (this.context.graphDocument.asyncGraphMode) {
-        // goto async node of trace
-        let asyncNode;
-        if (trace) {
-          const { applicationId } = trace;
-          const dp = allApplications.getById(applicationId).dataProvider;
-          const { rootContextId } = trace;
-          asyncNode = dp.indexes.asyncNodes.byRoot.getFirst(rootContextId);
-          if (this.followMode && asyncNode) {
-            await this.waitForAsyncGraph();
-            await this.remote.focusAsyncNode(asyncNode, ignoreFailed);
-          }
-        }
-        else {
-          this.clearFocus();
-        }
-
-        if (asyncNode) {
-          await this.waitForAsyncGraph();
-          await this.remote.selectAsyncNode(asyncNode, ignoreFailed);
-        }
-        else {
-          await this.remote.selectAsyncNode(null);
-        }
-      }
-      else {
-        let contextNode;
-        if (trace) {
-          const { applicationId, contextId } = trace;
-          contextNode = await this.getContextNodeById(applicationId, contextId);
-          if (this.followMode && contextNode) {
-            // NOTE: since we do this right after init, need to check if contextNode have been built
-            await this.focus(contextNode);
-          }
-        }
-
-        // always decorate ContextNode
-        this._selectContextNode(contextNode);
-      }
+      await this.owner.graph.handleTraceSelected?.(trace);
     }
     catch (err) {
       logTrace(`Cannot focus on selected trace #${trace?.traceId}`, trace, err);
     }
   }
 
-  handleHiddenNodeChanged = () => {
-    this.handleTraceSelected();
-  }
-
-  _selectContextNode(contextNode) {
-    if (this._selectedContextNode && !this._selectedContextNode.isDisposed) {
-      // deselect old
-      this._selectedContextNode.setSelected(false);
-    }
-
-    if (contextNode) {
-      // select new
-      contextNode.setSelected(true);
-    }
-
-    this._selectedContextNode = contextNode;
-  }
-
-  toggleFollowMode() {
-    const mode = !this.followMode;
-    this.setFollowMode(mode);
-    return mode;
-  }
-
-  setFollowMode(mode) {
-    if (this.followMode === mode) {
-      return;
-    }
-    this.followMode = mode;
-    if (this.followMode) {
-      this.handleTraceSelected();
-    }
-    else {
-      // this.lastHighlighter?.dec();
-      // this.lastHighlighter = null;
-    }
-    this._emitter.emit('modeChanged', this.followMode);
-  }
-
   /**
    * @param {ContextNode} node 
    */
   async focus(node) {
-    // if node is hidden, we focus on the hiddenNode
-    const hiddenNode = node.isHiddenBy();
+    // if node is hidden, focus on the hiddenNode instead
     let targetNode = node;
+    const hiddenNode = node.isHiddenBy?.();
```

Comment:
`isHiddenBy` indicates a `bool`, but this is a `Node`; name change recommended - maybe `hiddenByNode`?